activestorage/lib/active_storage/engine.rb

```diff
@@ -34,22 +34,21 @@ class Engine < Rails::Engine # :nodoc:
     end
 
     initializer "active_storage.services" do
-      config.after_initialize do |app|
-        if config_choice = app.config.active_storage.service
-          config_file = Pathname.new(Rails.root.join("config/storage.yml"))
-          raise("Couldn't find Active Storage configuration in #{config_file}") unless config_file.exist?
-
-          require "yaml"
-          require "erb"
-
-          configs =
-            begin
-              YAML.load(ERB.new(config_file.read).result) || {}
-            rescue Psych::SyntaxError => e
-              raise "YAML syntax error occurred while parsing #{config_file}. " \
-                    "Please note that YAML must be consistently indented using spaces. Tabs are not allowed. " \
-                    "Error: #{e.message}"
-            end
+      config.to_prepare do
+        if config_choice = Rails.configuration.active_storage.service
+          configs = Rails.configuration.active_storage.service_configurations ||= begin
```

Comment:
Maybe it is better to avoid the memoization and conditionally do everything, even the `to_prepare` hook? We could take advantage of the scope of the blocks to even avoiding have to memoize things or create a new config.

```ruby
    initializer "active_storage.services" do
      if config_choice = app.config.active_storage.service
        config_file = Pathname.new(Rails.root.join("config/storage.yml"))
        raise("Couldn't find Active Storage configuration in #{config_file}") unless config_file.exist?

        require "yaml"
        require "erb"

        configs =
          begin
            YAML.load(ERB.new(config_file.read).result) || {}
          rescue Psych::SyntaxError => e
            raise "YAML syntax error occurred while parsing #{config_file}. " \
              "Please note that YAML must be consistently indented using spaces. Tabs are not allowed. " \
              "Error: #{e.message}"
          end

        config.to_prepare do |app|
          ActiveStorage::Blob.service =
            begin
              ActiveStorage::Service.configure config_choice, configs
            rescue => e
              raise e, "Cannot load `Rails.config.active_storage.service`:\n#{e.message}", e.backtrace
            end
        end
      end
    end
```